In [1]:
import pandas as pd

In [118]:
data = pd.read_csv("dec/data/fulls/Declarations_third_splitted_table.csv", sep=';', dtype={'dest_edrpou': object})
data.head(5)

/home/roman/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3020: DtypeWarning: Columns (5,6,7,9) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


No                                         trash_name         code  \
0   1  Масла та мастила моторні, трансмісійні нехлоро...  6000.2.8.09   
1   2  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
2   3  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
3   4                    Підметини та пух хутра штучного  1830.2.1.06   
4   5  Плівка чи оболонка на основі полімерів зіпсова...  1753.1.1.03   

              alter_name class storage   value value_prediction  \
0  Відпрацьовані мастила    ІІ       0   0.012            0.012   
1     Промащене ганчір’я   ІІІ       0   0.100            0.211   
2        Побутове сміття    ІV       0  18.000            84.18   
3      Підметини з цехів    ІV       0  35.200             40.5   
4    Відходи поліетилену    ІV       0   1.912            2.000   

                                         destination value_destination  \
0  38563155 ТОВ «Науково- дослідний інститут «УКР...             0.012   
1  38563155 ТОВ «Науково- дослідний інститут «УКР...             0.100   
2  38641065 КП «Чисте місто» Жовтоводської місько...            18.000   
3  38563155 ТОВ «Науково- дослідний інститут «УКР...            35.200   
4  38563155 ТОВ «Науково- дослідний інститут «УКР...             1.912   

  operation operation_desc                  decl_num dest_edrpou  \
0        R9            NaN  D1221855500_00001_260716    38563155   
1        R5            NaN  D1221855500_00001_260716    38563155   
2        D1            NaN  D1221855500_00001_260716    38641065   
3        R5            NaN  D1221855500_00001_260716    38563155   
4        R5            NaN  D1221855500_00001_260716    38563155   

                                         dest_name  \
0  ТОВ «Науково- дослідний інститут «УКРЕКОПРОЕКТ»   
1  ТОВ «Науково- дослідний інститут «УКРЕКОПРОЕКТ»   
2     КП «Чисте місто» Жовтоводської міської ради"   
3  ТОВ «Науково- дослідний інститут «УКРЕКОПРОЕКТ»   
4  ТОВ «Науково- дослідний інститут «УКРЕКОПРОЕКТ»   

                                        dest_address  \
0  69002, м. Запоріжжя, вул. Червоногвардійська, ...   
1  69002, м. Запоріжжя, вул. Червоногвардійська, ...   
2          52200,м. Жовті Води, вул. Петровського,56   
3  69002, м. Запоріжжя, вул. Червоногвардійська, ...   
4  69002, м. Запоріжжя, вул. Червоногвардійська, ...   

                           data_fact_address   dest_date  
0  69000, м. Запоріжжя, вул. Теплична, 27. ,  08.01.2014  
1  69000, м. Запоріжжя, вул. Теплична, 27. ,  08.01.2014  
2                                        NaN         NaN  
3    69000, м. Запоріжжя, вул. Теплична, 27.         NaN  
4    69000, м. Запоріжжя, вул. Теплична, 27.         NaN

In [194]:
data2 = pd.read_csv('Декларації на 11-01-2018.csv', sep='\t')
data2['code'] = data2['code'].apply(lambda x: x.strip())
data2.at[1, 'code'] = '2802008594'
data2.head(5)

id                                               name        code  region  \
0   1                                      ПП "АІС І Ко"    22299407       1   
1   2                  ФОП ЧЕРНИШОВ АНДРІЙ ВОЛОДИМИРОВИЧ  2802008594       1   
2   3  ДЕРЖАВНЕ ПІДПРИЄМСТВО "КЕРЧЕНСЬКИЙ МОРСЬКИЙ РИ...    00467873       1   
3   4                                ТОВ "ЕКОСЕРВІСГРУП"    37978753       1   
4   5           ПРИВАТНЕ ПІДПРИЄМСТВО "КРИМ-ЕКОГІДРОТЕХ"    34406463       1   

                                            location list_of_jobs  \
0  95047, АР Крим, м.Сімферополь, вул. Вузлова, б...        1,2,3   
1  98300, АР Крим, м. Керч, вул. Шевченко, б. 56,...        1,2,3   
2   98320, АР Крим, м. Керч, вул. Свердлова, буд.49         1,3,4   
3  АР Крим, м. Сімферополь, вул. Балаклавська, бу...        1,3,7   
4  98300, АР Крим, м. Керч, вул. Орджонікідзе, бу...        1,3,5   

                                        list_of_vidh     license  \
0  Відцходи, що містять як складові або забруднюв...  12.12.2011   
1  Відпрацьовані нафтопродукти, не придатні для в...  25.11.2010   
2  Відходи сумішей масло/вода, вуглеводні/вода, е...  14.06.2011   
3  Відходи виробництва, одержання і застосування ...  28.12.2012   
4  Відпрацьовані нафтопродукти, не придатні для в...  16.10.2013   

  licensing_date license_start_period license_end_period  \
0     2011-12-12           0000-00-00         0000-00-00   
1     2010-11-25           0000-00-00         0000-00-00   
2     2011-06-14           0000-00-00         0000-00-00   
3     2012-12-28           0000-00-00         0000-00-00   
4     2013-10-16           0000-00-00         0000-00-00   

               string_region  is_licensee  confirmed license_status  \
0  Автономна Республіка Крим            1          1    Необмежений   
1  Автономна Республіка Крим            1          1    Необмежений   
2  Автономна Республіка Крим            1          1    Необмежений   
3  Автономна Республіка Крим            1          1    Необмежений   
4  Автономна Республіка Крим            1          1    Необмежений   

  reason_of_change  blocked  
0              NaN        0  
1              NaN        0  
2              NaN        0  
3              NaN        0  
4              NaN        0

In [26]:
name = 'Оксан'
code = '22299407'

In [17]:
names = []
for i in data2['name']:
    if not data.loc[data['dest_name'] == i].empty:
        names.append(i)
#     print(data.loc[data['dest_name'] == i])

In [20]:
data2.loc[data2['name'] == names[0]]

id                 name      code  region  \
3   4  ТОВ "ЕКОСЕРВІСГРУП"  37978753       1   

                                            location list_of_jobs  \
3  АР Крим, м. Сімферополь, вул. Балаклавська, бу...        1,3,7   

                                        list_of_vidh     license  \
3  Відходи виробництва, одержання і застосування ...  28.12.2012   

  licensing_date license_start_period license_end_period  \
3     2012-12-28           0000-00-00         0000-00-00   

               string_region  is_licensee  confirmed license_status  \
3  Автономна Республіка Крим            1          1    Необмежений   

  reason_of_change  blocked  
3              NaN        0

In [79]:
data.loc[data['dest_name'] == names[0]]

No                                         trash_name         code  \
17494   4  Матеріали фільтрувальні зіпсовані, відпрацьова...  7730.3.1.05   

                                              alter_name class storage value  \
17494  Матеріали фільтрувальні зіпсовані, відпрацьова...   ІІІ     0.0  0.02   

      value_prediction                                        destination  \
17494             0.02  37978753 ТОВ "ЕКОСЕРВІСГРУП", АР Крим, м. Сімф...   

      value_destination operation operation_desc                  decl_num  \
17494              0.02       R13            NaN  D5320410100_00048_080217   

      dest_edrpou            dest_name  \
17494    37978753  ТОВ "ЕКОСЕРВІСГРУП"   

                                            dest_address data_fact_address  \
17494  АР Крим, м. Сімферополь, вул. Балаклавська, бу...               NaN   

        dest_date  
17494  28.12.2012

In [191]:
code = '2802008594'
data.loc[data['dest_edrpou'] == code]
data2.loc[data2['code'] == code]
# data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:'].loc[data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'] == code]

Empty DataFrame
Columns: [0, 1, decl_num]
Index: []

In [159]:
data[data['dest_name'].str.lower().str.contains("Оксана".lower())]#['dest_name'].iloc[5]

No                trash_name         code alter_name class storage  \
98654   8  Відходи деревини кускові  2000.2.2.01      Дрова    ІV       0   

        value value_prediction  \
98654  57.631              120   

                                             destination value_destination  \
98654  3229313943 СПД ФО Миколенко Оксана Сергіївна, ...            57.631   

      operation operation_desc                  decl_num dest_edrpou  \
98654        R1            NaN  D5310136400_00210_270218  3229313943   

                               dest_name  \
98654  СПД ФО Миколенко Оксана Сергіївна   

                                            dest_address data_fact_address  \
98654  36000 м. Полтава, вул. Конституції, буд. 13, к...               NaN   

      dest_date  
98654       NaN

In [53]:
data['dest_name'].str.lower().str

In [59]:
data2.iloc[2]

id                                                                      3
name                    ДЕРЖАВНЕ ПІДПРИЄМСТВО "КЕРЧЕНСЬКИЙ МОРСЬКИЙ РИ...
code                                                             00467873
region                                                                  1
location                 98320, АР Крим, м. Керч, вул. Свердлова, буд.49 
list_of_jobs                                                        1,3,4
list_of_vidh            Відходи сумішей масло/вода, вуглеводні/вода, е...
license                                                        14.06.2011
licensing_date                                                 2011-06-14
license_start_period                                           0000-00-00
license_end_period                                             0000-00-00
string_region                                   Автономна Республіка Крим
is_licensee                                                             1
confirmed                             

In [70]:
data.loc[185539]['dest_edrpou']

'03328327'

In [82]:
codes = []
for i in data2['code']:
    if i in list(data['dest_edrpou']):
        codes.append(i)
#     print(data.loc[data['dest_name'] == i])

In [83]:
len(names)

4580

In [84]:
len(codes)

5665

In [108]:
data[data['dest_edrpou'].isin(list(data2['code'])) == False]

No                                         trash_name         code  \
42       1  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
43       2  Шлам від очищення вод стічних комунальних (міс...  9030.2.9.05   
79       3                                      Шлам септиків  7720.3.1.02   
80       4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
162      5                                    Послід пташиний  0124.2.6.03   
428      8               Відходи тканин рослинного походження  0111.2.6.02   
429      9  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
571      7                           Відходи деревини кускові  2000.2.2.01   
572      8                            Обапіл (крім шпального)  2000.2.2.02   
573      9                                    Стружка деревна  2000.2.2.09   
574     10                                    Тирса деревинна  2000.2.2.17   
575     11  Залишки плит деревностружкових та деревноволок...  2000.2.2.22   
4263     3  Шини, зіпсовані перед початком експлуатації, в...  6000.2.9.03   
5807     3  Шини, зіпсовані перед початком експлуатації, в...  6000.2.9.03   
6727     1       Батареї свинцеві зіпсовані або відпрацьовані  6000.2.9.04   
6728     2  Залишки очищення резервуарів для зберігання, щ...  6000.2.9.17   
6729     3  Масла та мастила моторні, трансмісійні інші зі...  6000.2.8.10   
6730     4    Відходи перевезень, не позначені іншим способом  6000.2.9.22   
6731     5  Відходи, стабілізовані чи затверділі за допомо...  9010.2.3.02   
6732     6  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
6738     5  Залишки очищення резервуарів для зберігання, щ...  6000.2.9.17   
6740     7  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
10238    3  Масла та мастила моторні, трансмісійні інші зі...  6000.2.8.10   
14125    2       Батареї свинцеві зіпсовані або відпрацьовані  6000.2.9.04   
14130    7  Шини, зіпсовані перед початком експлуатації, в...  6000.2.9.03   
14131    8  Деревина та вироби з деревини зіпсовані або ви...  7710.3.1.10   
14132    9  Суміш відходів, матеріалів та виробів з пластм...  7710.3.1.05   
14133   10                    Макулатура паперова та картонна  7710.3.1.01   
14238    1  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
14240    3  Відходи, стабілізовані чи затверділі за допомо...  9010.2.3.02   
...     ..                                                ...          ...   
186053   6  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186070   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186089   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186093   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186106   5  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186125  11  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186129   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186145   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186210   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186214   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186244  10                 Залишки зернові від очищення зерна  1561.2.9.04   
186302   2                 Пил від процесів оброблення каменю  2670.2.9.03   
186303   3  Камінь для пам’ятників та будівництва і вироби...  2670.3.1.01   
186314   6                 Залишки зернові від очищення зерна  1561.2.9.04   
186328   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186332   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186336   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186340   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186398   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186426   8  Продукти харчові з

In [110]:
data2[data2['code'].isin(list(data['dest_edrpou'])) == False]

In [88]:
data['dest_edrpou'].unique().shape

(5361,)

In [89]:
data

No                                         trash_name         code  \
0        1  Масла та мастила моторні, трансмісійні нехлоро...  6000.2.8.09   
1        2  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
2        3  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
3        4                    Підметини та пух хутра штучного  1830.2.1.06   
4        5  Плівка чи оболонка на основі полімерів зіпсова...  1753.1.1.03   
5        6                    Макулатура паперова та картонна  7710.3.1.01   
6        7                        Одяг зношений чи зіпсований  7710.3.1.13   
7        1                                          Бій цегли  2640.2.9.02   
8        2                                      Шлак паливний  4010.2.8.01   
9        1                          Вичавки плодові та ягідні  1532.2.9.02   
10       2  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
11       1  Лампи люмінесцентні та відходи, які містять рт...  7710.3.1.26   
12       2       Батареї свинцеві зіпсовані або відпрацьовані  6000.2.9.04   
13       3  Масла та мастила моторні, трансмісійні інші зі...  6000.2.8.10   
14       4  Шини, зіпсовані перед початком експлуатації, в...  6000.2.9.03   
15       5                 Брухт чорних металів дрібний інший  7710.3.1.08   
16       6                                    Тирса деревинна  2000.2.2.17   
17       7                           Відходи деревини кускові  2000.2.2.01   
18       8  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
19       1  Лампи люмінесцентні та відходи, які містять рт...  7710.3.1.26   
20       2                           Пил полірувальних кругів  2681.2.9.02   
21       3  Матеріали обтиральні зіпсовані, відпрацьовані ...  7730.3.1.06   
22       4                        Одяг зношений чи зіпсований  7710.3.1.13   
23       5  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
24       1  Акумулятори лужні, у т. ч. кадмій-нікелеві, не...  3120.3.1.04   
25       2  Лампи люмінесцентні та відходи, які містять рт...  7710.3.1.26   
26       3  Мастила (суміші спирто-бензинові, масла мінера...  2910.1.0.05   
27       4  Грунти, забруднені нафтопродуктами, хімічними ...  4590.3.1.06   
28       5  Бій скла технічного та скловиробів, що не підл...  7710.3.1.03   
29       6  Тара металева використана, у т. ч. дрібна (бан...  7710.3.1.07   
...     ..                                                ...          ...   
186531   4  Туші тварин зіпсовані, забруднені або неіденти...  1511.1.1.04   
186532   5  Оболонки для ковбас зіпсовані, забруднені або ...  1513.1.1.08   
186533   6  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186534   1  Батареї та акумулятори інші зіпсовані або відп...  6000.2.9.08   
186535   2  Масла та мастила моторні, трансмісійні інші зі...  6000.2.8.10   
186536   3  Сировина та матеріали допоміжні інші зіпсовані...  0111.1.2.04   
186537   4  Шини, зіпсовані перед початком експлуатації, в...  6000.2.9.03   
186538   5                 Брухт чорних металів дрібний інший  7710.3.1.08   
186539   6                    Макулатура паперова та картонна  7710.3.1.01   
186540   7  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186541   1  Відходи інші, збирання та знищення яких обумов...  8510.2.9.06   
186542   2  Обладнання та інструменти медичні одноразові з...  8510.2.9.02   
186543   3       Тара аптекарська зіпсована або відпрацьована  8530.2.9.04   
186544   4  Відходи комунальні (міські) змішані, у т. ч. с...  7720.3.1.01   
186545   1  Лампи люмінесцентні та відходи, які містять рт...  7710.3.1.26   
186546   2  Батареї та акумулятори інші зіпсовані або відп...  6000.2.9.08   
186547   3  Масла та мастила моторні, трансмісійні інші зі...  6000.2.8.10   
186548   4  Абсорбенти зіпсовані, відпрацьовані чи забруднені  7730.3.1.04   
186549   5  Матеріали фільтрувальні зіпсовані, відпрацьова...  7730.3.1.05   
186550   6  Матеріали обтираль

### Edrpou in data_all: 186561(unique 5361)
### Edrpou in data2018: 6062(unique 4025)
### Edrpou in data_all but not in data2018: 5158(1668 unique)
### Edrpou in data2018 but not in data_all: 388(332 unique)

In [106]:
for i in list(set(cd3)):
    print(i.strip())

04055222
36055819
30860372
02082717
2809504030
25599771
32443702
38351440
244701268
35314287
04379730
33219415
03361135
22299407
08588791
37747817
32216654
32940150
00203826
32570366
12345896
33389678
34848301
32940156
112201453
24658075
35369501
32655402
37744088
37485972
36707878
04329613
38510323
30003330
38113641
31269993
2809504032
02970174
2434311653
04366413
32756601
37450316
13353875
153698799
12345689
242467778
00201514
33665604
0002970174
01994586
35165938
04345718
04348800
00152230
03768767
37233014
32577340
36841065
36193683
23280093
40002202
08588599
03355293
24463880
32099653
34435049
31880284
00191052
23552742
39193691
32479875
5310900000
21858509
33153784
2561704495
2766500610
2974409010
34808274
30297024
37360998
00698041
37997400
4823355702
30069040
51012375
20364201
19329547
05453516
00195452
29419041
30829259
20485152
37761082
31141049
30659913
32419720
2312500290
14312973
32464937
12345678
23089646
39409474
5
32730091
374565390
36934047
2071818243
37425426
03756000

In [116]:
len(data['dest_edrpou'].unique())

5361

In [121]:
data1 = pd.read_csv("dec/data/fulls/Declarations_first_table.csv", sep=';')#, dtype={'dest_edrpou': object})
data1.head(9)

0  \
0            Назва суб'єкта господарювання:   
1                      Код згідно з ЄДРПОУ:   
2                      Код згідно з КОАТУУ:   
3  Юридична адреса суб’єкта господарювання:   
4                Електронна пошта /Телефон:   
5            Назва суб'єкта господарювання:   
6                      Код згідно з ЄДРПОУ:   
7                      Код згідно з КОАТУУ:   
8  Юридична адреса суб’єкта господарювання:   

                                                   1                  decl_num  
0      Товариство з обмеженою відповідальністю «ФИМ»  D1221855500_00001_260716  
1                                           05468386  D1221855500_00001_260716  
2                                         1221855500  D1221855500_00001_260716  
3  53081,Дніпропетровська обл.Криворізький район,...  D1221855500_00001_260716  
4                       dir@fim.dp.ua/(05652)3-36-40  D1221855500_00001_260716  
5  Товариство з обмеженою відповідальністю "Оксан...  D7321586001_00001_100616  
6                                           31878951  D7321586001_00001_100616  
7                                         7321586001  D7321586001_00001_100616  
8  59433, Чернівецька обл., Заставнівський р-н, с...  D7321586001_00001_100616

In [126]:
data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'].shape

(21926,)

In [130]:
data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'].unique().shape

(8110,)

In [148]:
cd77 = data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:'][data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'].isin(list(data2['code'])) == False]['1']

In [140]:
cd66 = data2[data2['code'].isin(list(data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'])) == False]['code'].unique().shape

In [138]:
data2[data2['code'].isin(list(data['dest_edrpou']) + list(data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'])) == False]['code'].unique().shape

(306,)

In [149]:
for i in cd77 :
    if i not in list(set(cd2)):
        print(i)

05468386
31878951
36748170
05398487
23077206
30162340
31571594
19472552
30990262
33002104
33651058
40199749
40199749
40199749
2156200422
37948133
14242681
31834542
34721796
22821513
37479985
00308376
00852909
08681086
08564943
37298747
36870831
31314574
30877035
36861518
05480619
33167561
37440905
24836720
30424720
36265925
36265925
36265925
36265925
36265925
36265925
2626309918
24607936
30299152
32166939
32019054
13449167
34973979
24607913
31677100
24607913
2622700672
24607936
2048218227
0024679315
0025683299
0025683365
0025683419
0020551587
0025683388
0002683394
0000131564
0025683336
0025683313
0025683322
0025683342
0025683081
0025683402
0025683307
24607936
0025683425
24607936
0025683371
0025683359
24607936
0033541383
0003361046
36906202
0038162272
35315676
0040081195
36018997
00334793
23730020
01564897
00849304
31806913
37954970
38346864
02969248
14312364
38199598
24925429
32069550
37815085
22928671
39424712
01412785
02011930
21631756
01993701
25288858
39908375
37633636
31815960
318

37243279
38281336
23685704
38384098
36265925
31504952
35771853
32193229
14307311
21642228
36265925
04824844
2649117050
23149176
20437874
01998414
31946015
13457623
36564953
36270081
01273036
05385571
22842751
30698245
30911290
31344923
32988737
30763036
31344923
37243279
26156320
03792852
36750263
01997768
00131529
00131529
23990212
00131529
00131529
2189806060
2115620418
14226133
3159826498
33665597
23972496
00380505
00131529
33296918
32234448
14353083
22699035
30684913
32894618
2723004115
32894618
13954534
2903104735
32890660
31898936
08734581
13307734
37738624
01526454
37776078
37138877
25879878
36551064
32404375
2528013136
2528013136
2528013136
05468386
24705171
22837974
32386477
04013867
33369342
22767506
24710344
08564854
20423205
30823262
34378714
36518687
22453340
01882503
01561887
31207130
20007889
32049199
32635098
38479802
36851782
00131529
00131529
32579002
00131529
20426043
00131529
32446629
96800921
00131529
36939217
31052661
20415128
38215153
35348880
2626309918
35025914

03359701
33437815
26156403
24783018
2197111802
37948133
03361402
02969188
34688885
20915842
39749568
32562528
00687310
26156461
40527818
31043704
31847843
02004999
31909570
00147134
32319924
01188052
34323974
32653583
00447356
3011914914
02005603
39580475
01983068
30841082
00993372
31405650
03791841
30688237
34154698
25180724
33976979
31405650
36345607
03446529
00231024
03119530
21067735
30841082
00953318
31405650
38916558
32570408
30841082
26038098
36066802
38916558
34852077
36595337
30485824
30841082
24607907
38916558
39184446
30841082
30200202
30841082
38916558
21738520
38916558
30841082
38916558
38916558
34430873
38916558
38916558
38916558
39580475
38916558
38916558
30072467
05502381
32075024
00383679
02941406
35365643
05484126
37854674
2554908422
20467378
39117830
2687004650
23343582
24420066
25483595
03117033
32447293
02007578
40982829
05490701
32550381
22153980
33264625
34962841
40982829
39685082
30109129
39685082
40982829
39685082
32902141
13813756
33467483
23543186
39225450
24

37157500
23244621
38981517
13731068
30006132
38512192
37122545
13303765
01181877
30024481
03731276
25236666
37806217
22440001
34880895
34880895
34532720
00479712
30424720
37066465
03794696
05408680
13731923
32049199
25236666
32049199
32594083
13734034
25184774
34019124
30161290
34363300
38428192
03288817
03328646
04330728
01181877
01181877
02001305
20609105
33679183
30221510
25181787
36293177
05784785
34342384
01527497
33884915
20112362
39661246
36626595
0032490244
00450045
30487093
38305367
19201907
01181877
01181877
22794087
37930503
33404067
08564943
20551995
00991551
02006596
00292497
31821381
31450441
31821381
24885150
31821381
30059069
35779941
31821381
13705191
2976300249
03057976
25560823
31982410
32049199
03348809
32837916
01994391
02649880
05491600
31941174
34596103
02005007
38305367
38305367
38305367
32654257
32654257
32654257
35535544
32654257
39685082
40380506
32654257
32654257
36299535
32654257
03735435
32654257
23889527
32654257
32654257
21665011
13706150
33065515
326542

25204531
05461390
31405650
34019061
23712217
30889103
20608169
32513287
25418539
36490830
2200013239
23955204
32393437
19482929
31405650
21203457
00955963
30797746
01992376
32792184
32792184
22189570
22191070
01111115
24331167
32792184
03580251
23905301
05405842
00446374
37278275
37278275
37278275
32577471
39675980
13977624
35512066
24924140
33911120
38010015
31958392
30641031
23744453
23744453
23744453
24596990
33680115
24596990
24596990
24596990
22764778
01994126
32628976
00203134
00203594
05407746
38281687
31855391
00200217
32738132
32738132
30583064
32738132
32245423
05484445
32700465
31749798
40921090
35411322
30603572
39195699
30603572
39207955
30603572
30603572
30603572
24475392
30603572
30603572
32950656
30603572
31268455
39195788
39195463
36304293
39502491
39200640
19424369
36108100
13723792
2928215035
34524882
33715891
24905094
13701069
34069039
35876116
32445705
14173336
34355758
33894121
20623068
30651825
38085452
20009931
22957595
37278275
32905273
40315877
32090645
350130

32869822
34244416
34634447
31200837
35728295
35615975
02223767
36265925
01994936
1706606492
35336429
01182500
03581434
31259168
32005345
14181927
05504811
31495420
08681139
2851804754
2626309918
01989591
30095458
40823143
32611978
24217371
32956692
35349747
31010092
31017171
23251963
02971759
32445705
30772868
23683680
36814630
34558576
32654257
23683740
34894494
32654257
31938863
31101839
36230520
31559121
32654257
32101850
32654257
31236643
30720689
13572598
31316807
25204577
00377163
40271442
32654257
31527875
36584550
24221705
31778022
32967502
37367218
37274037
31634381
00993417
00291411
31269862
00380787
31517060
00147134
21256123
04725987
30958103
13915675
00130659
13976731
23058539
05486792
31517060
01413891
05385571
08680170
33123608
38941212
03733513
34055755
31527875
00309536
40252545
01994592
21380685
22644502
31895338
30652737
14247520
05480619
30935374
39437758
31148107
33476189
32337278
35247968
19338138
32090645
00707857
00488941
36134497
01181618
36265925
13581352
3626

32257512
31083108
2160712391
00374479
37792058
41005969
05388233
30675610
31740030
31320734
25302854
22837974
36828617
03446558
21377133
32789941
31320656
05460441
19282308
36265925
37667056
30146910
14308032
14181442
22699101
36265925
05460441
32894618
33271386
05460441
00306756
30651825
32167106
01984263
24596990
34323508
32789894
31007137
38580013
05461071
22670637
34173314
32594083
03356884
37144375
03446529
03749129
33065515
34498219
32843212
05393518
32478720
00156630
21665011
03749106
35917412
25717070
23972496
36751707
14027095
30697262
27735041
36200559
00131529
25483595
30444510
2910712631
33665597
35563545
00292497
05518871
31450441
30149623
36370552
32409127
33710516
22219803
22153980
30275304
32578653
34515141
22701074
32902141
33945448
05417228
32319924
30147542
31855391
33584049
33341835
00131529
00381574
39006240
32049199
38614230
36584456
1997714025
04654336
25593544
00131529
34531124
40059699
38326992
32191331
32049199
35423668
33491799
33392209
38528093
34063587
1925

19326017
31576901
38740702
13507918
36584037
37978278
31427962
32444952
34329981
31176910
38916558
38916558
00991551
14111810
25468796
38916558
31963989
00238167
03754478
26156538
05784785
38916558
00377791
04854874
38916558
20880196
30957225
36987468
13360705
02008951
33559260
41276682
39749568
39755505
32166939
39534104
32730400
03729871
14331367
31878815
40472785
26156461
03534593
40982829
33933762
35534844
35206539
21606540
40982829
40982829
40982829
38740702
38740702
38740702
38740702
38740702
38740702
38740702
38740702
38740702
38740702
33054482
38305791
40725552
26464748
20446778
20455228
25204695
05795783
33648312
33519080
35741279
30405890
23898624
03580251
39223296
35642878
38740702
41264153
30913099
02005757
30825081
08563783
02780232
33768964
2365212651
32932993
21328928
04543542
20551995
32655926
40082324
41762459
25418539
05502462
00215812
14173394
40982829
03370737
34336627
05387179
05461303
30510703
33520384
00157842
35774461
31570412
22828432
22210369
32301189
25380435

20518650
30794745
39608851
25180724
00203571
35161687
30398460
34772809
40982829
40982829
24419910
00204607
31380537
14309729
00310093
39755505
32416841
33680115
34430873
34430873
34430873
34430873
34430873
40024865
37815064
08371339
19461235
36828617
2651820270
34545593
31599101
00955963
38384339
32566213
02494973
02003445
02082522
32468926
33120853
03581888
32638083
24171048
34596103
21013142
31526164
02002730
03755452
30880624
32638214
04012365
02071175
37709860
22699070
30402329
00444263
05432684
32150288
01598360
14279843
35505844
39781142
24981630
04527885
21952144
34688885
2197111802
30754807
01372850
30841082
03732146
2576611053
32884395
01998383
37915045
36035551
2397201845
30841082
03755348
30841082
30841082
30841082
39608851
21203457
32276142
39608851
24331167
23240407
31648478
30841082
30841082
30841082
34313222
37399852
30841082
30841082
35659242
22690927
14103790
40105417
33328882
31828113
28635036
23744453
23744453
23744453
23744453
22089851
30841082
33844657
30841082
23

21665318
0358143
36427583
00242909
36169844
32536452
39145889
05408668
25277777
34403295
13548581
35848260
25208960
34558576
30136085
24357511
39921574
38580872
32306522
05477066
14307570
35025914
32445705
22957595
34894494
32101850
01561887
32874151
03328468
30630145
30944991
03328468
33608589
39634724
39634808
21685172
14361581
30161127
33818826
36029050
2688413799
33884915
30430356
30161127
36293177
21174655
32504424
00707857
21690995
31826484
30334673
32925902
36595337
36647332
25186483
39148222
14346268
2362000545
24210334
19414617
20635255
00447238
04718013
03058655
31889628
14310968
21672519
24667246
05407344
2160712391
20056770
39907319
16476839
30466356
23524007
03771442
16472861
25298093
32713673
24265186
22967501
35291194
36824984
37000975
30453389
30535885
32678388
14287650
21055620
40039390
30797746
22070231
31941530
40478106
33775566
30650753
33286633
40531379
02648709
24733539
02782314
30361225
35049694
13925644
34173314
35508919
31992915
03187648
2362900016
24221562
391

3124120475
21560766
34430873
19313492
30691606
23244621
21560766
30650679
34430873
21560766
22487238
2424607670
34430873
2470212570
00293640
14271932
31517060
39006240
39770371
20784943
38705799
37282674
00721780
23390156
31929492
36420474
23867078
41033708
03450301
39189837
31929492
00310628
41540066
23058539
31929492
22034017
36230520
31929492
34323508
3019614311
33672303
34244416
00954586
31929492
01204348
31929492
13674102
38305367
24099262
31929492
3266405366
31929492
03449083
03115815
31929492
00309134
31929492
01897908
30425482
38305367
31929492
35250849
31929492
19347315
31043704
03359090
31929492
02495431
32917247
21489100
31929492
04012661
30142062
03359753
05761850
00293278
31929492
32134630
31929492
03746384
25586053
31929492
22946057
33096208
32231515
31439454
31558600
40060636
14205651
32013005
00191891
01529122
34082463
38305367
33282218
22837974
34161733
02219406
36330737
30559953
19420704
31133719
36577538
34470937
25204695
20851817
25204560
31881723
21560766
33651791


04529370
24710344
32049199
31778022
34288102
25204608
01994451
37738624
32086437
23082976
32049199
30482268
24712426
04737111
25613367
33648312
24220002
42369400
39011491
03450436
32531054
01994971
35634228
23756522
31922460
24764713
05415473
24102142
22372966
03764850
01598301
02403446
37102560
23906565
23756522
31690582
23744453
34063482
05761850
25381316
13471936
13929625
42412838
36334993
40585670
30072493
23900634
33165873
3126604955
2283814787
2230122724
30568287
37260027
30088241
00147134
32730400
02011054
29109054
14281020
20780043
23756522
14346989
31576901
24413020
30838462
02007578
05513715
00711971
20591306
1773403159
14257180
30104493
22794124
37278275
34709632
32301189
20054481
05445563
00307402
02969188
02008951
32049199
00992763
34442700
33296133
36328293
22372280
35039953
25180150
24676707
34341375
31522243
30767690
20188299
34329981
36564953
21370764
31522243
41105190
26362083
31522243
32049199
37758846
26435031
22644502
38607087
32395764
39718285
21561949
00210996
37

38321216
38321216
30386370
34771119
38321216
30841082
20084335
30841082
30841082
38732288
31014894
39502491
25385857
38321216
36241582
39195699
39195463
30616508
2617919140
38321216
21173466
01993701
02470135
35954611
31678214
24932292
32911376
34933255
32244959
30982361
30804386
40657853
2239815510
35059414
30982361
30982361
32421280
22927134
05417325
30982361
24370724
30982361
30982361
19256073
25745327
01882048
31729918
35519082
34063592
24939503
36173190
30743491
34825841
32811490
31987554
00130872
21336490
30982361
14282829
03733772
38567563
19130857
30982361
01565112
30982361
30982361
21462905
34218716
01412710
05456779
19022659
32830051
03058715
30384724
34218716
34218716
24723387
33894121
39134348
41104967
36703318
41104967
31035740
32607495
41104967
20616217
30739569
41104967
23761121
39221545
03114783
01977731
05484126
06654613
03730590
36424573
36003865
37023353
13428745
35101874
25434969
35101874
14063926
00740599
30426664
22089851
32483848
19165985
23759880
20502092
305545

In [158]:
data1.loc[5]['1']

'Товариство з обмеженою відповідальністю "Оксана - М"'

In [188]:
code = '00467873'
data.loc[data['dest_edrpou'] == code]
data2.loc[data2['code'] == code]
# data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:'].loc[data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'] == code]

Empty DataFrame
Columns: [0, 1, decl_num]
Index: []

In [172]:
data2.loc[data2['code'] == code]

id                     name      code  region  \
279    280   ТОВ " Укрресурси-2011"  37923407      26   
2818  3074  ДП"Фірма Альфатер Київ"  37923407      26   

                                               location list_of_jobs  \
279   03127, м. Київ, вул. Героїв Оборони, буд. 9/10...        1,2,3   
2818                            п-р.Повітрофлотський 72          NaN   

                                           list_of_vidh     license  \
279   Відходи, що містять як складові або забруднюва...  15.04.2013   
2818                                                NaN         NaN   

     licensing_date license_start_period license_end_period string_region  \
279      2013-04-15           0000-00-00         0000-00-00    Місто Київ   
2818     0000-00-00           0000-00-00         0000-00-00           NaN   

      is_licensee  confirmed license_status reason_of_change  blocked  
279             1          1    Необмежений              NaN        0  
2818            0          1    Необмежений              NaN        0

In [178]:
data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:'].loc[data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'] == code]
data1[23150:23155]

0  \
23150            Назва суб'єкта господарювання:   
23151                      Код згідно з ЄДРПОУ:   
23152                      Код згідно з КОАТУУ:   
23153  Юридична адреса суб’єкта господарювання:   
23154                Електронна пошта /Телефон:   

                                                      1  \
23150                     ТОВ " Богдан - Авто Чернівці"   
23151                                          37923407   
23152                                        7310136900   
23153           58007 м. Чернівці, вул. Коломийська, 2Г   
23154  mykola.pasichnyk@bogdanauto.com.ua/(0372) 586828   

                       decl_num  
23150  D7310136900_00165_280217  
23151  D7310136900_00165_280217  
23152  D7310136900_00165_280217  
23153  D7310136900_00165_280217  
23154  D7310136900_00165_280217

In [174]:
# data2[(data2['code'].isin(list(data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'])) and data2['code'].isin(list(data['dest_edrpou'])))]
# data[data['dest_edrpou'].isin(data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:'][data1.loc[data1['0'] == 'Код згідно з ЄДРПОУ:']['1'].isin(list(data2['code']))]['1'])]['dest_edrpou']

In [195]:
data_liz = pd.read_excel('Ліцензійний реєстр (поводження з небезпечними відходами).xlsx', dtype={'ИНН': object})
del data_liz['Unnamed: 3']
del data_liz['Unnamed: 8']
data_liz

ИНН                                           39458830  \
0    2802008594                  ФОП ЧЕРНИШОВ АНДРІЙ ВОЛОДИМИРОВИЧ   
1      00186536  ПРИВАТНЕ АКЦІОНЕРНЕ ТОВАРИСТВО ЕЛЕКТРОМЕТАЛУРГ...   
2      00190928  Акціонерне товариство ПОКРОВСЬКИЙ ГІРНИЧО-ЗБАГ...   
3      00191052          ПРАТ ДРУЖКІВСЬКИЙ ЗАВОД МЕТАЛЕВИХ ВИРОБІВ   
4      00191230  ПАТ ЗАПОРІЗЬКИЙ МЕТАЛУРГІНИЙ КОМБІНАТ ЗАПОРІЖС...   
5      00195452  ПУБЛІЧНЕ АКЦІОНЕРНЕ ТОВАРИСТВО АРТЕМІВСЬКИЙ ЗА...   
6      00201514  ДОНЕЦЬКИЙ ДЕРЖАВНИЙ НАУКОВО-ДОСЛІДНИЙ ТА ПРОЕК...   
7      00203826                                           ПАТ АЗОТ   
8      00467873  ДЕРЖАВНЕ ПІДПРИЄМСТВО КЕРЧЕНСЬКИЙ МОРСЬКИЙ РИБ...   
9      01125815         ДП ІЗМАЇЛЬСЬКИЙ МОРСЬКИЙ ТОРГОВЕЛЬНИЙ ПОРТ   
10     02004500  Хмельницький обласний протитуберкульозний дисп...   
11     05393079                                         ПрАТ ЄВРАЗ   
12     05393085              ПрАТ ДНІПРОВСЬКИЙ КОКСОХІМІЧНИЙ ЗАВОД   
13     05416768  РИТУАЛЬНА СЛУЖБА СПЕЦІАЛІЗОВАНЕ КОМУНАЛЬНЕ ПІД...   
14     05761264         ПРАТ ШОСТКИНСЬКИЙ ЗАВОД ХІМІЧНИХ РЕАКТИВІВ   
15     05763814               ПАТ КРЮКІВСЬКИЙ ВАГОНОБУДІВНИЙ ЗАВОД   
16     08588599  АВАРІЙНО-РЯТУВАЛЬНИЙ ЗАГІН СПЕЦІАЛЬНОГО ПРИЗНА...   
17     08588688  АВАРІЙНО-РЯТУВАЛЬНИЙ ЗАГІН СПЕЦІАЛЬНОГО ПРИЗНА...   
18     08588725  Аварійно-рятувальний загін спеціального призна...   
19     08588791  АВАРІЙНО-РЯТУВАЛЬНИЙ ЗАГІН СПЕЦІАЛЬНОГО ПРИЗНА...   
20     08588843     УПРАВЛІННЯ МНС УКРАЇНИ В МИКОЛАЇВСЬКІЙ ОБЛАСТІ   
21     08588961  ГОЛОВНЕ УПРАВЛІННЯ МНС УКРАЇНИ В ХАРКІВСЬКІЙ О...   
22     08588990  АВАРІЙНО РЯТУВАЛЬНИЙ ЗАГІН СПЕЦІАЛЬНОГО ПРИЗНА...   
23     13500129  МАЛЕ ПІДПРИЄМСТВО У ФОРМІ ТОВ  ДОСЛІДНЕ ВИРОБН...   
24     13685821             МАЛЕ ПІДПРИЄМСТВО У ФОРМІ ТОВ  СОЛЯРІС   
25     13736978                                           МПП РАДА   
26     13818222                         Приватне підприємство РІАЛ   
27     14045437                         ПП МАЛЕ ПІДПРИЄМСТВО АЛЬФА   
28     14047092                 МАЛЕ ПРИВАТНЕ ПІДПРИЄМСТВО ФОРТУНА   
29     14307794                                      ПАТ МОТОР СІЧ   
..          ...                                                ...   
323    39115938                                     ТОВ СП-ЕКОНІКА   
324  2312500290                  ФОП Соколовський Олег Миколайович   
325  2421501713                    ФОП БІЛОУС СЕРГІЙ ВОЛОДИМИРОВИЧ   
326  2430602117                   ФОП ПРОКОПЮК ВІТАЛІЙ ОЛЕКСІОЙВИЧ   
327    13744541               ТОВ НАУКОВО-ВИРОБНИЧА ФІРМА ЕКОЦЕНТР   
328    32576064                               ТОВ ЕКОЛОГІЯ УКРАЇНА   
329    20077743  ДЕРЖАВНЕ ПІДПРИЄМСТВО З ПИТАНЬ ПОВОДЖЕННЯ З ВІ...   
330    03351898  КРЕМЕНЧУЦЬКЕ КОМУНАЛЬНЕ АВТОТРАНСПОРТНЕ ПІДПРИ...   
331    39193691                       ТОВ КАЛУСЬКА СОЛЯНА КОМПАНІЯ   
332    38454023                                         ТОВ ІН ЕКО   
333    38140443                                          ТОВ АРГАН   
334    38788257                                     ТОВ ЕКОХОЛДИНГ   
335    37642215                                 ТОВ УТІЛІС ІННОТЕХ   
336    39409474                                   ТОВ АНАТОМІЯ ЕКО   
337    34406463                ПРИВАТНЕ ПІДПРИЄМСТВО  ЕКОГІДРОТЕХ    
338    32821513                                      ПП ГРАНД ЛАЙТ   
339    00152230                         ПАТ НАФТОХІМІК ПРИКАРПАТТЯ   
340    33252506                                 ТОВ ПРОМСНАБ-ЛЬВІВ   
341    31730834                                  ТОВ КОМПАНІЯ Т.К.   
342    00131305          ПУБЛІЧНЕ АКЦІОНЕРНЕ ТОВАРИСТВО КИЇВЕНЕРГО   
343    00017733                ПАТ СУДНОПЛАВНА КОМПАНІЯ УКРРІЧФЛОТ   
344    14312973                    ВАТ МЕРИДІАН ім. С.П. Корольова   
345    30146931                                         ПП ОЛЕСТАС   
346    38510323                  ТОВ "СУЧАСНІ ЕКОЛОГІЧНІ ІНОВАЦІЇ"   
347    35001301                                 ТОВ МІДАС-КОМПАНІЯ   
348    05607824       

In [198]:
data2.shape

(6062, 17)